pkg/kubelet/prober/prober.go

```diff
@@ -274,6 +274,26 @@ func (eic execInContainer) SetStderr(out io.Writer) {
 	//unimplemented
 }
 
+func (eic execInContainer) SetEnv(env []string) {
+	//unimplemented
+}
+
 func (eic execInContainer) Stop() {
 	//unimplemented
 }
+
+func (eic execInContainer) Start() error {
+	return fmt.Errorf("unimplemented")
+}
+
+func (eic execInContainer) Wait() error {
+	return fmt.Errorf("unimplemented")
+}
+
+func (eic execInContainer) StdoutPipe() (io.ReadCloser, error) {
+	return nil, fmt.Errorf("unimplemented")
+}
+
+func (eic execInContainer) StderrPipe() (io.ReadCloser, error) {
+	return nil, fmt.Errorf("unimplemented")
+}
```

Comment:
`execInContainer` needs to implement any methods that are called by the exec prober.

Reply:
In my initial exploration it looked easier as it is, I got a bit lost in the different abstractions.
I think, as [outlined in the other comment][c], we need to do an async `Exec` instead of an `ExecSync`. That's the important piece. If and how this bubbles up to this `execInContainer` struct is a different story and the easy part.


[c]: https://github.com/kubernetes/kubernetes/pull/71316/files#r256769402